### PR TITLE
MINOR: adopt java8 features and cleanup

### PR DESCRIPTION
**Pull request by levzem**
Signed-off-by: Lev Zemlyanov <lev@confluent.io>

## Problem
the connector still uses java7 style when it can be updated to rely on java8 features
TPWTest was in the wrong package

## Solution
move TPWTest to the root package and and replace java7 language features with java8

##### Does this solution apply anywhere else?
- [ ] yes
- [x] no

## Test Strategy
run existing tests

##### Testing done:
- [x] Unit tests
- [x] Integration tests
- [ ] System tests
- [ ] Manual tests

### CODE REVIEW

File: src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java [hunk 1]
```diff
@@ -874,46 +873,37 @@ private void setRetryTimeout(long timeoutMs) {
   }
 
   private void createHiveTable() {
```
Comment (dosvath) [inline src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java:875]: Do we have tests covering `createHiveTable()`, `alterHiveSchema()` and `addHivePartition()`, or verified the intended behavior of the updated lambda expressions other ways?

**Review by dosvath — COMMENTED** [review]
Thanks @levzem, minor question.

File: src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java (hunk 1 above)
Comment (levzem) [reply src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java:875]: yes, we have multiple tests like these https://github.com/confluentinc/kafka-connect-hdfs/blob/master/src/test/java/io/confluent/connect/hdfs/avro/HiveIntegrationAvroTest.java
for each each dataformat

**Review by dosvath — APPROVED** [review]
LGTM

**Pull request merged**